src/mame/casio/pv1000.cpp

```diff
@@ -89,40 +102,63 @@ void pv1000_sound_device::voice_w(offs_t offset, uint8_t data)
 
 
 /*
- plgDavid's audio implementation/analysis notes:
+	plgDavid's audio implementation/analysis notes:
 
- Sound appears to be 3 50/50 pulse voices made by cutting the main clock by 1024,
- then by the value of the 6bit period registers.
- This creates a surprisingly accurate pitch range.
+	Sound appears to be 3 50/50 pulse voices made by cutting the main clock by 1024,
+	then by the value of the 6bit period registers.
+	This creates a surprisingly accurate pitch range.
 
- Note: the register periods are inverted.
- */
+	Note: the register periods are inverted.
+
+	plgDavid 2023 update: lidnariq (NESDEV) took a fondness to the system and gave me a bunch of test roms 
+	to strenghten the emulation. 
+	
+	Quite a few things were uncovered overall, but for audio specifically:
+	1)Audio mix/mux control ($FB, case 0x03) was ignored
+	2)lidnariq's tracing of my PCB scans showed that all three sound outputs are mixed using different volumes:
+	  square1 via i/o$F8 is -6dB, square2 via i/o$F9 is -3dB, defining square3 via i/o$FA as 0dB
+*/
 
 void pv1000_sound_device::sound_stream_update(sound_stream &stream, std::vector<read_stream_view> const &inputs, std::vector<write_stream_view> &outputs)
 {
 	auto &buffer = outputs[0];
 
-	for (int sampindex = 0; sampindex < buffer.samples(); sampindex++)
-	{
-		s32 sum = 0;
-
-		for (int i = 0; i < 3; i++)
-		{
-			uint32_t per = (0x3f - (m_voice[i].period & 0x3f));
+	//Each channel has a different volume via resistor mixing which correspond to -6dB, -3dB, 0dB drops
+	static const int volumes[3] = {0x1000, 0x1800, 0x2000};
 
-			if (per != 0)   //OFF!
-				sum += m_voice[i].val * 8192;
+	for (int index = 0; index < buffer.samples(); index++){
+		s32 sum = 0;
```

Comment:
Once again, please follow the existing formatting convention – use Allman style in this file.

Reply:
Should be done.